Lib/javascript/v8/javascriptcode.swg

```diff
@@ -377,8 +377,10 @@ fail:
 %{
   /* Name: $jsmangledname, Type: $jsmangledtype, Dtor: $jsdtor */
   SWIGV8_FUNCTION_TEMPLATE $jsmangledname_class = SWIGV8_CreateClassTemplate("$jsmangledname");
-  SWIGV8_SET_CLASS_TEMPL($jsmangledname_clientData.class_templ, $jsmangledname_class);
-  $jsmangledname_clientData.dtor = $jsdtor;
+  if ($jsmangledname_clientData.class_templ.IsEmpty()) {
+    SWIGV8_SET_CLASS_TEMPL($jsmangledname_clientData.class_templ, $jsmangledname_class);
+    $jsmangledname_clientData.dtor = $jsdtor;
+  }
```

Comment:
This code goes into the SWIGV8_INIT init function. I think you've put this in because it is possible for the init  function to be called more than one?? If so, it feels like it needs to be thread-safe and this change would not solve a race condition. There would also need to be a ton more work done in the init function as a lot of it operates on global variables.

Reply:
> If so, it feels like it needs to be thread-safe

That's `isolate->Enter();` and `isolate->Exit();`'s job. I mean those are responsible for serialization of complete init function, so that we can safely check for all of IsEmpty-s.